app/controllers/challenge_signups_controller.rb

```diff
@@ -322,4 +323,54 @@ def prompt_meme_to_csv
 
     csv_array
   end
+
+  private
+
+  def challenge_signup_params
+    params.require(:challenge_signup).permit(
+      :pseud_id,
+      requests_attributes: nested_prompt_params,
+      offers_attributes: nested_prompt_params
+    )
+  end
+
+  def nested_prompt_params
+    [
+      :id,
+      :collection_id,
+      :title,
+      :url,
+      :any_fandom,
+      :any_character,
+      :any_relationship,
+      :any_freeform,
+      :any_category,
+      :any_rating,
+      :any_warning,
+      :anonymous,
+      :description,
+      :_destroy,
+      tag_set_attributes: [
+        :id,
```

Comment:
I'm having a hard time determining why this is necessary, but the tests definitely fail without it. Someone else want to double check this and see what I'm missing? (The tests do pass if `id` is removed from `tag_set_attributes` in the prompts controller, though.)